### PR TITLE
RichDEM build_tarballs.jl

### PR DESCRIPTION
**Pull request by sunayana**
Wrapping of the C++ RichDEM package from :
https://github.com/Cervest/richdem
From release 2.3.1: https://github.com/Cervest/richdem/releases/tag/v2.3.1

### CODE REVIEW

**Comment by sunayana** [comment]
Would it be possible for someone to cancel this build since I would like to update the link to the sources to be build.

File: R/RichDEM/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    ArchiveSource(
+        "https://github.com/Cervest/richdem/archive/refs/tags/v$(version).zip",
+        "6c87d1fa4c417b7f518c3f2964a1688d2e2f74b6b1381270dc38d741ec709db5",
+    ),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd richdem-2.3.1/
```
Comment (giordano) [inline R/RichDEM/build_tarballs.jl:19]:
```suggestion
cd $WORKSPACE/srcdir/richdem-*
```

**Review by giordano — COMMENTED** [review]
You didn't change the URL, did you?

**Comment by giordano** [comment]
There's no need to keep updating on `master`, please don't do it, that operation only adds more and more jobs to the queue.

**Comment by sunayana** [comment]
Thanks a lot @giordano

**Comment by sunayana** [comment]
> There's no need to keep updating on `master`, please don't do it, that operation only adds more and more jobs to the queue.

sorry will keep that in mind.

**Comment by sunayana** [comment]
> You didn't change the URL, did you?

No I did not, I verified it was the latest.

**Comment by sunayana** [comment]
Since this is my first time doing this, is `RichDEM_jll.jl` package  automatically created and available in JuliaBinaryWrappers namespace after this PR is merged?

**Comment by giordano** [comment]
Yes, correct!  You'll be automatically pinged in the pull request opened in https://github.com/JuliaRegistries/General to add the package to the registry.  Thanks for the package BTW!

**Comment by sunayana** [comment]
> Yes, correct! You'll be automatically pinged in the pull request opened in https://github.com/JuliaRegistries/General to add the package to the registry. Thanks for the package BTW!

Thank you !

**Pull request merged**